ts/Core/Series/OverlappingDataLabels.ts

```diff
@@ -106,15 +106,30 @@ function chartHideOverlappingLabels(
                     x: label.attr('x'),
                     y: label.attr('y')
                 },
-                bBox = label.getBBox();
+                bBox = label.getBBox(),
+                alignValue = label.alignValue || '',
+                width = bBox.width;
 
-            label.width = bBox.width;
+            let alignOffset = 0;
+
+            label.width = width;
             label.height = bBox.height;
 
+            switch (alignValue) {
+                case 'right':
+                    alignOffset = width;
+                    break;
+                case 'center':
+                    alignOffset = (width / 2);
+                    break;
+                default:
+                    break;
+            }
```

Comment:
Use the [getAlignFactor](https://github.com/highcharts/highcharts/blob/2f11abca284057d324d5fb58b8ecce897bebb803/ts/Core/Utilities.ts#L1371-L1376) function